app/models/manageiq/providers/autosde/storage_manager/autosde_client.rb

```diff
@@ -70,7 +70,9 @@ def login
     data = self.AuthenticationApi.token_auth_post(auth_request, opts)
     @token = data.token
   rescue => e
-    raise AUTH_ERRR_MSG + e.to_s
+    error_data = JSON.parse(e.response_body)
```

Comment:
Thanks for the PR @eilam20 !
Only issue I see here is that this `rescue` block will catch any exception type, and while `e.to_s` will work with any type of exception `e.response_body` will not.

Can you catch a more specific type of exception like `rescue AutosdeOpenapiClient::ApiError => e` for parsing the response_body, then leave this "fallback" exception handler the way it is?